### PR TITLE
fix: Add e2e test for routing forms webhook subscription

### LINKED ISSUE #9038: [CAL-1737] Write FORM_SUBMITTED webhook e2e test
We have tests to verify eventType webhooks but not the Routing Form webhook

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1737](https://linear.app/calcom/issue/CAL-1737/write-form-submitted-webhook-e2e-test)</sub>

### PR DESCRIPTION
**Pull request by tachyonicbytes**
## What does this PR do?

Add e2e test for routing forms webhook subscription
I also moved the `gotoRoutingLink` function used for routing form testing into the `@calcom/web/playwright/lib/testUtils` module

Fixes #9038 

## Type of change

- Chore (add e2e test)

## How should this be tested?

- [x] Run the e2e test: yarn test-e2e webhook -g "FORM_SUBMITTED"

Please let me know if there are more asserts to add.

/claim #9038

### CODE REVIEW

**Comment by usmaann1** [comment]
/claim #9038 

Test Plan: End-to-End Testing for Routing Forms Webhook Subscription

Objective:
The objective of this test plan is to perform end-to-end testing for the routing forms webhook subscription. The test plan includes both manual tests and automated tests to ensure comprehensive verification. The focus will be on running the e2e test using the specified command and verifying the functionality of the gotoRoutingLink function.

Test Scope:
The test scope includes executing the e2e test specifically for the webhook subscription of routing forms. The test plan will cover the functionality of the gotoRoutingLink function after its relocation to the @calcom/web/playwright/lib/testUtils module.

Test Environment:
The test environment should consist of:

The application or system where the routing forms webhook subscription is implemented.
Access to the routing forms and their webhook configuration.
Test data or content to simulate the submission of a routing form.
Test Plan:
4.1 Manual Test:

4.1.1 Manual Verification of Routing Forms Webhook Subscription:
Objective: To manually verify the functionality of the routing forms webhook subscription.
Steps:

- Identify the routing forms that are configured with webhook subscriptions.
- Generate test data or content that represents a valid submission for a routing form.
- Submit the test data through the routing form.
- Monitor the webhook endpoint or logs to confirm that the webhook is triggered.
- Verify that the expected data from the form submission is received by the webhook endpoint.

4.1.2 Manual Testing of gotoRoutingLink Function:
Objective: To manually test the gotoRoutingLink function after its relocation to the @calcom/web/playwright/lib/testUtils module.

Steps:
- Identify the test scenarios or conditions in which the gotoRoutingLink function is expected to be invoked.
- Manually trigger the gotoRoutingLink function with the appropriate parameters in each identified scenario.
- Verify that the function successfully navigates to the expected routing link.
- Confirm that the routed page or destination corresponds to the expected behavior.

4.2 Automated Test:

4.2.1 End-to-End Testing for Routing Forms Webhook Subscription:
Objective: To automate the end-to-end testing of the webhook subscription for routing forms.
Steps:

- Develop an e2e test script specifically for testing the webhook subscription functionality.
- Configure the test script to simulate the submission of a routing form with test data.
- Ensure that the test script triggers the webhook endpoint.
- Verify that the expected data from the form submission is received by the webhook endpoint.
- Implement assertions in the test script to validate the correctness of the received data.
- Execute the e2e test script using the specified command: yarn test-e2e webhook -g "FORM_SUBMITTED".
- Review the test results and ensure that all assertions pass, indicating the successful webhook subscription functionality.

Test Reporting:
For each test, document the observed results, any deviations from expected behavior, and any defects found during testing. Report and track the issues found using a bug-tracking system.

Test Completion Criteria:
The test execution is considered complete when:

- The manual verification of the webhook subscription for routing forms has been performed and documented.
- The manual testing of the gotoRoutingLink function has been executed, and the results are documented.
- The e2e test for the webhook subscription using the specified command has been executed, and the test results are documented.

**Review by CarinaWolli — DISMISSED** [review]
Tests and type checks are failing because `saveCurrentForm` function got deleted

File: packages/app-store/routing-forms/playwright/tests/basic.e2e.ts [hunk 1]
```diff
@@ -594,35 +595,3 @@ async function selectNewRoute(page: Page, { routeSelectNumber = 1 } = {}) {
...
-    previewLink = `/forms/${formId}`;
-  }
-
-  await page.goto(`${previewLink}${queryString ? `?${queryString}` : ""}`);
-
-  // HACK: There seems to be some issue with the inputs to the form getting reset if we don't wait.
-  await new Promise((resolve) => setTimeout(resolve, 500));
-}
-
-async function saveCurrentForm(page: Page) {
```
[the remark this answers is not in the document]
Comment (tachyonicbytes) [reply packages/app-store/routing-forms/playwright/tests/basic.e2e.ts:625]: Thanks for the tip! On it

File: apps/web/playwright/webhook.e2e.ts [hunk 2]
```diff
@@ -387,3 +388,58 @@ test.describe("BOOKING_REQUESTED", async () => {
     webhookReceiver.close();
   });
 });
+
+test.describe("FORM_SUBMITTED", async () => {
+  test("can submit a form and get a submission event", async ({ page, users }) => {
+    const webhookReceiver = createHttpServer();
+    const user = await users.create();
+
+    await user.login();
```
Comment (roae) [inline apps/web/playwright/webhook.e2e.ts:397]: this should be 
```suggestion
    await user.apiLogin();
```

File: packages/app-store/routing-forms/playwright/tests/basic.e2e.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tachyonicbytes) [reply packages/app-store/routing-forms/playwright/tests/basic.e2e.ts:625]: Done

**Comment by tachyonicbytes** [comment]
The pipeline passes now.

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**